File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-fixed-size/distinct-elements-window/code/java/DistinctElementsWindowSol1.java
class Solution {
    public int[] distinctInWindow(int[] nums, int k) {
        int[] res = new int[nums.length-k+1];
        for (int i=0;i<=nums.length-k;i++) {
            Set<Integer> s = new HashSet<>();
            for (int j=i;j<i+k;j++) s.add(nums[j]);
            res[i] = s.size();
        } return res;
    }
}

File: src/data/courses/dsa/two-pointers-sliding-window/sliding-window-fixed-size/distinct-elements-window/code/cpp/distinct_elements_window_sol1.cpp
vector<int> distinctInWindow(vector<int>& nums, int k) {
    vector<int> result;
    for (int i = 0; i <= (int)nums.size()-k; i++) {
        unordered_set<int> s(nums.begin()+i, nums.begin()+i+k);
        result.push_back(s.size());
    }
    return result;
}